app/src/main/java/com/example/qstreak/viewmodels/ProfileViewModel.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.qstreak.viewmodels
+
+import android.content.SharedPreferences
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.qstreak.db.UserRepository
+import com.example.qstreak.network.ApiResult
+import com.example.qstreak.utils.UID
+import com.example.qstreak.utils.USER_NAME
+import com.example.qstreak.utils.USER_ZIP
+import kotlinx.coroutines.launch
+
+class ProfileViewModel(
+    private val sharedPrefs: SharedPreferences,
+    private val userRepository: UserRepository,
+    private val sharedPrefsEditor: SharedPreferences.Editor
+) : ViewModel() {
+    val userName = MutableLiveData<String>(sharedPrefs.getString(USER_NAME, null))
+    val userZipCode = MutableLiveData<String>(sharedPrefs.getString(USER_ZIP, null))
+    val nameError = MutableLiveData<Boolean>(false)
+    val zipCodeError = MutableLiveData<Boolean>(false)
+    val profileUpdated = MutableLiveData<Boolean>(false)
+
+    private val uid: String? by lazy {
+        sharedPrefs.getString(UID, null)
+    }
+
+    // Used in display of Edit Profile Name/Zip Code screens
+    val errorToDisplay = MutableLiveData<String>()
+
+    fun saveProfile() {
+        val name = userName.value?.let { it } ?: ""
+        val zipCode = userZipCode.value
+
+        if (!validateInputs(name, zipCode)) {
+            return
+        }
+
+        try {
+            viewModelScope.launch {
+                val response =
+                    userRepository.updateUser(name, zipCode!!, uid.toString())
+
+                if (response is ApiResult.Success) {
+                    sharedPrefsEditor.putString(USER_NAME, name).commit()
+                    sharedPrefsEditor.putString(USER_ZIP, zipCode).commit()
+                    refreshNameAndZipCode()
+                    profileUpdated.value = true
+                } else {
+                    errorToDisplay.value = (response as ApiResult.Error).apiErrors
+                }
+            }
+        } catch (e: Exception) {
+            errorToDisplay.value = e.message
+        }
+    }
+
+    private fun refreshNameAndZipCode() {
```

Comment:
Is this method needed anymore? Those values have already been updated by the time this is called, right?

Reply:
the values are updated in shared prefs but not in the view model itself (lines 46-48 here)